kornia/augmentation/container/ops.py

```diff
@@ -286,10 +287,11 @@ def transform_list(
         if isinstance(module, (K.GeometricAugmentationBase2D,)):
             tfm_input = []
             params = cls.get_instance_module_param(param)
-            for inp in input:
-                inp = module.transform_tensor(inp)
+            params_i = copy.deepcopy(params)
```

Comment:
Why do we need to make a copy here ?

Reply:
To iteratively modify the batch_prob.

In batched cases, image would a tensor of shape (B, C, H, W) whereas masks would be a list of tensors [(N1, H, W), (N2, H, W)....].

`params["batch_prob"]` is set w.r.t to the image tensor, so we need to iterate over it and reassign the batch_prob passed while keeping the rest of the params the same

Reply:
I think there should at least have an `if-else` block here. If it is a list, we enumerate them one-by-one. If it is a straight-away tensor, we do it all at once.

Reply:
Not sure I follow. This change takes place in the `transform_list` function of `MaskSequentialOps`. So the input is a list of mask tensors.

Reply:
Ah, this should be fine then.